### ISSUE TITLE
Frequent crashes when training a model with large number of groups.

### ISSUE THREAD

**Issue by mathmanu** [NVIDIA/caffe#425]
In the configuration that I am trying to train, there are a large number groups - in fact every group has a size of 4. I am observing frequent crashes. The iteration of crash is not fixed .The training abruptly stops without any warning or error. 

The behavior is same with CUDNN6 and CUDNN7. The crash occurs even if I set:
default_cnv_algos_override: "1,1,1"

This lead me to believe that this could be a caffe issue and not a CUDNN issue. 

I have attached a zip file with configuration files and logs.
[imagenet_2017-10-05_21-58-39_crash.zip](https://github.com/NVIDIA/caffe/files/1363812/imagenet_2017-10-05_21-58-39_crash.zip)

Thankyou for your support.

**Comment by mathmanu** [comment]
@drnikolaev wonder if you could look into this issue quickly. I get a crash in approximately about 30,000 iterations - sometimes less, sometimes more. Thanks.

**Comment by drnikolaev** [comment]
Hi @mathmanu while I'm looking to this, could you try my staging branch please? https://github.com/drnikolaev/caffe/tree/caffe-0.16
Also, logs in the zip appear to be truncated, no crash call stack in them.

**Comment by mathmanu** [comment]
Sure I'll try that and let you know. The logs are as it is generated - there is no call stack generated - it just crashes silently.

**Comment by mathmanu** [comment]
While trying to compile https://github.com/drnikolaev/caffe/tree/caffe-0.16, I get the following compilation error.

CXX src/caffe/data_transformer2.cpp
src/caffe/data_transformer2.cpp:23:6: error: prototype for \u2018void caffe::DataTransformer<Dtype>::TransformGPU(const caffe::TBlob<Dtype>*, caffe::TBlob<Dtype>*, const std::array<unsigned int, 3ul>&, bool)\u2019 does not match any in class \u2018caffe::DataTransformer<Dtype>\u2019
 void DataTransformer<Dtype>::TransformGPU(const TBlob<Dtype>* input_blob, TBlob<Dtype>* transformed_blob,
      ^
In file included from src/caffe/data_transformer2.cpp:10:0:
./include/caffe/data_transformer.hpp:32:8: error: candidate is: void caffe::DataTransformer<Dtype>::TransformGPU(int, int, int, int, size_t, const void*, Dtype*, const unsigned int*)
   void TransformGPU(int N, int C, int H, int W, size_t sizeof_element, const void* in, Dtype* out,
        ^
src/caffe/data_transformer2.cpp:35:6: error: prototype for \u2018void caffe::DataTransformer<Dtype>::Transform(const caffe::TBlob<Dtype>*, caffe::TBlob<Dtype>*, const std::array<unsigned int, 3ul>&, bool)\u2019 does not match any in class \u2018caffe::DataTransformer<Dtype>\u2019
 void DataTransformer<Dtype>::Transform(const TBlob<Dtype>* input_blob, TBlob<Dtype>* transformed_blob,

**Comment by mathmanu** [comment]
I am able to build https://github.com/drnikolaev/caffe/tree/caffe-0.16. Will be trying out now.

More information:
Initially I got the following link error:
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/4.8/../../../x86_64-linux-gnu/libturbojpeg.a(libturbojpeg_la-turbojpeg.o): relocation R_X86_64_32 against `.data' can not be used when making a shared object; recompile with -fPIC
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../x86_64-linux-gnu/libturbojpeg.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
make: *** [.build_release/lib/libcaffe-nv.so.0.16.4] Error 1

Then I applied the fix described here and the linking worked:
https://github.com/OpenKinect/libfreenect2/issues/36

sudo apt-get install libturbojpeg
sudo ln -s /usr/lib/x86_64-linux-gnu/libturbojpeg.so.0.0.0 /usr/lib/x86_64-linux-gnu/libturbojpeg.so

Not sure why this link error is not observed in NVIDIA/caffe but only in drnikolaev/caffe. Even in drnikolaev/caffe, it was not there earlier.

**Comment by mathmanu** [comment]
As the training started, I observed that drnikolaev/caffe is about 15% faster compared to NVIDIA/caffe. Interesting.

**Comment by drnikolaev** [comment]
Hi @mathmanu yes, I did some performance improvements (including groups support).
As of libturbojpeg, this is known packaging bug, CMake handles it properly, but regular Make can't. Creating missing softlink is the right way.
Are crashes gone?

**Comment by mathmanu** [comment]
drnikolaev/caffe also crashed. No call stack is dumped. It just silently crashes like NVIDIA/caffe.

**Comment by drnikolaev** [comment]
Could you run `ldd` against your `caffe.so` please?

**Comment by mathmanu** [comment]
Attached the ldd outputs after running:
ldd libcaffe-nv.so

[NVIDIA_caffe_ldd(libcaffe-nv.so).txt](https://github.com/NVIDIA/caffe/files/1371203/NVIDIA_caffe_ldd.libcaffe-nv.so.txt)
[drnikolaev_caffe_ldd(libcaffe-nv.so).txt](https://github.com/NVIDIA/caffe/files/1371205/drnikolaev_caffe_ldd.libcaffe-nv.so.txt)

**Comment by drnikolaev** [comment]
Well, our lists are pretty much different (see mine below). Is there any way to upgrade to boost 1.58 and CUDA 9?

```
	linux-vdso.so.1 =>  (0x00007ffd5b342000)
	libboost_system.so.1.58.0 => /usr/lib/x86_64-linux-gnu/libboost_system.so.1.58.0 (0x00007f3843a4b000)
	libboost_thread.so.1.58.0 => /usr/lib/x86_64-linux-gnu/libboost_thread.so.1.58.0 (0x00007f3843824000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f3843607000)
	libglog.so.0 => /usr/lib/x86_64-linux-gnu/libglog.so.0 (0x00007f38433d8000)
	libgflags.so.2 => /usr/lib/x86_64-linux-gnu/libgflags.so.2 (0x00007f38431b6000)
	libprotobuf.so.9 => /usr/lib/x86_64-linux-gnu/libprotobuf.so.9 (0x00007f3842e98000)
	libhdf5_serial.so.10 => /usr/lib/x86_64-linux-gnu/libhdf5_serial.so.10 (0x00007f38429fb000)
	libhdf5_serial_hl.so.10 => /usr/lib/x86_64-linux-gnu/libhdf5_serial_hl.so.10 (0x00007f38427da000)
	liblmdb.so.0 => /usr/lib/x86_64-linux-gnu/liblmdb.so.0 (0x00007f38425c5000)
	libleveldb.so.1 => /usr/lib/x86_64-linux-gnu/libleveldb.so.1 (0x00007f384236b000)
	libturbojpeg.so.0 => /usr/lib/x86_64-linux-gnu/libturbojpeg.so.0 (0x00007f3842109000)
	libcudart.so.9.0 => /usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudart.so.9.0 (0x00007f3841e9c000)
	libcurand.so.9.0 => /usr/local/cuda-9.0/targets/x86_64-linux/lib/libcurand.so.9.0 (0x00007f383df3a000)
	libcublas.so.9.0 => /usr/local/cuda-9.0/targets/x86_64-linux/lib/libcublas.so.9.0 (0x00007f383ab03000)
	libcudnn.so.7 => /usr/lib/x86_64-linux-gnu/libcudnn.so.7 (0x00007f3829b9a000)
	libopencv_highgui.so.2.4 => /usr/lib/x86_64-linux-gnu/libopencv_highgui.so.2.4 (0x00007f3829951000)
	libopencv_imgproc.so.2.4 => /usr/lib/x86_64-linux-gnu/libopencv_imgproc.so.2.4 (0x00007f38294c5000)
	libopenblas.so.0 => /opt/OpenBLAS/lib/libopenblas.so.0 (0x00007f3828536000)
	libpython2.7.so.1.0 => /usr/lib/x86_64-linux-gnu/libpython2.7.so.1.0 (0x00007f3827fa8000)
	libboost_python-py27.so.1.58.0 => /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.58.0 (0x00007f3827d5b000)
	libnccl.so.2 => /usr/lib/x86_64-linux-gnu/libnccl.so.2 (0x00007f381bd7f000)
	libnvidia-ml.so.1 => /usr/lib/nvidia-384/libnvidia-ml.so.1 (0x00007f381b7ac000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f381b4a2000)
	libopencv_core.so.2.4 => /usr/lib/x86_64-linux-gnu/libopencv_core.so.2.4 (0x00007f381b078000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f381acf6000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f381aadf000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f381a715000)
	/lib64/ld-linux-x86-64.so.2 (0x000055b1c1acd000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f381a50d000)
	libunwind.so.8 => /usr/lib/x86_64-linux-gnu/libunwind.so.8 (0x00007f381a2f1000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f381a0d7000)
	libsz.so.2 => /usr/lib/x86_64-linux-gnu/libsz.so.2 (0x00007f3819ed4000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f3819ccf000)
	libsnappy.so.1 => /usr/lib/x86_64-linux-gnu/libsnappy.so.1 (0x00007f3819ac7000)
	libGL.so.1 => /usr/lib/nvidia-384/libGL.so.1 (0x00007f3819822000)
	libjpeg.so.8 => /usr/lib/x86_64-linux-gnu/libjpeg.so.8 (0x00007f38195c9000)
	libpng12.so.0 => /lib/x86_64-linux-gnu/libpng12.so.0 (0x00007f38193a4000)
	libtiff.so.5 => /usr/lib/x86_64-linux-gnu/libtiff.so.5 (0x00007f381912f000)
	libjasper.so.1 => /usr/lib/x86_64-linux-gnu/libjasper.so.1 (0x00007f3818eda000)
	libIlmImf-2_2.so.22 => /usr/lib/x86_64-linux-gnu/libIlmImf-2_2.so.22 (0x00007f3818a0c000)
	libHalf.so.12 => /usr/lib/x86_64-linux-gnu/libHalf.so.12 (0x00007f38187c8000)
	libgtk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0 (0x00007f381817d000)
	libgdk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0 (0x00007f3817ec8000)
	libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007f3817c74000)
	libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f3817963000)
	libgtkglext-x11-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libgtkglext-x11-1.0.so.0 (0x00007f381775f000)
	libgdkglext-x11-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdkglext-x11-1.0.so.0 (0x00007f38174fa000)
	libdc1394.so.22 => /usr/lib/x86_64-linux-gnu/libdc1394.so.22 (0x00007f3817284000)
	libv4l1.so.0 => /usr/lib/x86_64-linux-gnu/libv4l1.so.0 (0x00007f381707e000)
	libavcodec-ffmpeg.so.56 => /usr/lib/x86_64-linux-gnu/libavcodec-ffmpeg.so.56 (0x00007f3815c51000)
	libavformat-ffmpeg.so.56 => /usr/lib/x86_64-linux-gnu/libavformat-ffmpeg.so.56 (0x00007f3815853000)
	libavutil-ffmpeg.so.54 => /usr/lib/x86_64-linux-gnu/libavutil-ffmpeg.so.54 (0x00007f38155e4000)
	libswscale-ffmpeg.so.3 => /usr/lib/x86_64-linux-gnu/libswscale-ffmpeg.so.3 (0x00007f3815354000)
	libtbb.so.2 => /usr/lib/x86_64-linux-gnu/libtbb.so.2 (0x00007f3815117000)
	libgfortran.so.3 => /usr/lib/x86_64-linux-gnu/libgfortran.so.3 (0x00007f3814deb000)
	libutil.so.1 => /lib/x86_64-linux-gnu/libutil.so.1 (0x00007f3814be8000)
	liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007f38149c5000)
	libaec.so.0 => /usr/lib/x86_64-linux-gnu/libaec.so.0 (0x00007f38147bd000)
	libGLX.so.0 => /usr/lib/nvidia-384/libGLX.so.0 (0x00007f381458d000)
	libGLdispatch.so.0 => /usr/lib/nvidia-384/libGLdispatch.so.0 (0x00007f38142be000)
	libjbig.so.0 => /usr/lib/x86_64-linux-gnu/libjbig.so.0 (0x00007f38140b0000)
	libIex-2_2.so.12 => /usr/lib/x86_64-linux-gnu/libIex-2_2.so.12 (0x00007f3813e91000)
	libIlmThread-2_2.so.12 => /usr/lib/x86_64-linux-gnu/libIlmThread-2_2.so.12 (0x00007f3813c8a000)
	libgmodule-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007f3813a86000)
	libpangocairo-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0 (0x00007f3813878000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f381353e000)
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007f3813338000)
	libatk-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0 (0x00007f3813112000)
	libcairo.so.2 => /usr/lib/x86_64-linux-gnu/libcairo.so.2 (0x00007f3812dfe000)
	libgdk_pixbuf-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0 (0x00007f3812bdc000)
	libgio-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0 (0x00007f3812853000)
	libpangoft2-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0 (0x00007f381263d000)
	libpango-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0 (0x00007f38123f1000)
	libfontconfig.so.1 => /usr/lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007f38121ad000)
	libXrender.so.1 => /usr/lib/x86_64-linux-gnu/libXrender.so.1 (0x00007f3811fa3000)
	libXinerama.so.1 => /usr/lib/x86_64-linux-gnu/libXinerama.so.1 (0x00007f3811d9f000)
	libXi.so.6 => /usr/lib/x86_64-linux-gnu/libXi.so.6 (0x00007f3811b8f000)
	libXrandr.so.2 => /usr/lib/x86_64-linux-gnu/libXrandr.so.2 (0x00007f3811984000)
	libXcursor.so.1 => /usr/lib/x86_64-linux-gnu/libXcursor.so.1 (0x00007f381177a000)
	libXcomposite.so.1 => /usr/lib/x86_64-linux-gnu/libXcomposite.so.1 (0x00007f3811576000)
	libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007f3811373000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007f3811161000)
	libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6 (0x00007f3810f58000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f3810ce8000)
	libGLU.so.1 => /usr/lib/x86_64-linux-gnu/libGLU.so.1 (0x00007f3810a78000)
	libXmu.so.6 => /usr/lib/x86_64-linux-gnu/libXmu.so.6 (0x00007f381085f000)
	libpangox-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangox-1.0.so.0 (0x00007f381063f000)
	libraw1394.so.11 => /usr/lib/x86_64-linux-gnu/libraw1394.so.11 (0x00007f381042f000)
	libusb-1.0.so.0 => /lib/x86_64-linux-gnu/libusb-1.0.so.0 (0x00007f3810217000)
	libv4l2.so.0 => /usr/lib/x86_64-linux-gnu/libv4l2.so.0 (0x00007f3810009000)
	libswresample-ffmpeg.so.1 => /usr/lib/x86_64-linux-gnu/libswresample-ffmpeg.so.1 (0x00007f380fdeb000)
	libva.so.1 => /usr/lib/x86_64-linux-gnu/libva.so.1 (0x00007f380fbcf000)
	libzvbi.so.0 => /usr/lib/x86_64-linux-gnu/libzvbi.so.0 (0x00007f380f944000)
	libxvidcore.so.4 => /usr/lib/x86_64-linux-gnu/libxvidcore.so.4 (0x00007f380f62f000)
	libx265.so.79 => /usr/lib/x86_64-linux-gnu/libx265.so.79 (0x00007f380ea10000)
	libx264.so.148 => /usr/lib/x86_64-linux-gnu/libx264.so.148 (0x00007f380e66c000)
	libwebp.so.5 => /usr/lib/x86_64-linux-gnu/libwebp.so.5 (0x00007f380e40f000)
	libwavpack.so.1 => /usr/lib/x86_64-linux-gnu/libwavpack.so.1 (0x00007f380e1e6000)
	libvpx.so.3 => /usr/lib/x86_64-linux-gnu/libvpx.so.3 (0x00007f380ddc2000)
	libvorbisenc.so.2 => /usr/lib/x86_64-linux-gnu/libvorbisenc.so.2 (0x00007f380db18000)
	libvorbis.so.0 => /usr/lib/x86_64-linux-gnu/libvorbis.so.0 (0x00007f380d8ec000)
	libtwolame.so.0 => /usr/lib/x86_64-linux-gnu/libtwolame.so.0 (0x00007f380d6c9000)
	libtheoraenc.so.1 => /usr/lib/x86_64-linux-gnu/libtheoraenc.so.1 (0x00007f380d489000)
	libtheoradec.so.1 => /usr/lib/x86_64-linux-gnu/libtheoradec.so.1 (0x00007f380d26f000)
	libspeex.so.1 => /usr/lib/x86_64-linux-gnu/libspeex.so.1 (0x00007f380d056000)
	libshine.so.3 => /usr/lib/x86_64-linux-gnu/libshine.so.3 (0x00007f380ce48000)
	libschroedinger-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libschroedinger-1.0.so.0 (0x00007f380cb73000)
	libopus.so.0 => /usr/lib/x86_64-linux-gnu/libopus.so.0 (0x00007f380c929000)
	libopenjpeg.so.5 => /usr/lib/x86_64-linux-gnu/libopenjpeg.so.5 (0x00007f380c705000)
	libmp3lame.so.0 => /usr/lib/x86_64-linux-gnu/libmp3lame.so.0 (0x00007f380c490000)
	libgsm.so.1 => /usr/lib/x86_64-linux-gnu/libgsm.so.1 (0x00007f380c282000)
	libcrystalhd.so.3 => /usr/lib/x86_64-linux-gnu/libcrystalhd.so.3 (0x00007f380c066000)
	libssh-gcrypt.so.4 => /usr/lib/x86_64-linux-gnu/libssh-gcrypt.so.4 (0x00007f380be1d000)
	librtmp.so.1 => /usr/lib/x86_64-linux-gnu/librtmp.so.1 (0x00007f380bc00000)
	libmodplug.so.1 => /usr/lib/x86_64-linux-gnu/libmodplug.so.1 (0x00007f380b875000)
	libgme.so.0 => /usr/lib/x86_64-linux-gnu/libgme.so.0 (0x00007f380b627000)
	libbluray.so.1 => /usr/lib/x86_64-linux-gnu/libbluray.so.1 (0x00007f380b3dd000)
	libgnutls.so.30 => /usr/lib/x86_64-linux-gnu/libgnutls.so.30 (0x00007f380b0ad000)
	libbz2.so.1.0 => /lib/x86_64-linux-gnu/libbz2.so.1.0 (0x00007f380ae9d000)
	libquadmath.so.0 => /usr/lib/x86_64-linux-gnu/libquadmath.so.0 (0x00007f380ac5d000)
	libfreetype.so.6 => /usr/lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007f380a9b3000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f380a790000)
	libpixman-1.so.0 => /usr/lib/x86_64-linux-gnu/libpixman-1.so.0 (0x00007f380a4e8000)
	libxcb-shm.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0 (0x00007f380a2e4000)
	libxcb-render.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-render.so.0 (0x00007f380a0d9000)
	libselinux.so.1 => /lib/x86_64-linux-gnu/libselinux.so.1 (0x00007f3809eb7000)
	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007f3809c9b000)
	libharfbuzz.so.0 => /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007f3809a3d000)
	libthai.so.0 => /usr/lib/x86_64-linux-gnu/libthai.so.0 (0x00007f3809834000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f380960a000)
	libXt.so.6 => /usr/lib/x86_64-linux-gnu/libXt.so.6 (0x00007f38093a1000)
	libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x00007f3809380000)
	libv4lconvert.so.0 => /usr/lib/x86_64-linux-gnu/libv4lconvert.so.0 (0x00007f3809106000)
	libsoxr.so.0 => /usr/lib/x86_64-linux-gnu/libsoxr.so.0 (0x00007f3808ea1000)
	libnuma.so.1 => /usr/lib/x86_64-linux-gnu/libnuma.so.1 (0x00007f3808c95000)
	libogg.so.0 => /usr/lib/x86_64-linux-gnu/libogg.so.0 (0x00007f3808a8c000)
	liborc-0.4.so.0 => /usr/lib/x86_64-linux-gnu/liborc-0.4.so.0 (0x00007f380880b000)
	libgcrypt.so.20 => /lib/x86_64-linux-gnu/libgcrypt.so.20 (0x00007f380852a000)
	libgssapi_krb5.so.2 => /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2 (0x00007f38082df000)
	libhogweed.so.4 => /usr/lib/x86_64-linux-gnu/libhogweed.so.4 (0x00007f38080ac000)
	libnettle.so.6 => /usr/lib/x86_64-linux-gnu/libnettle.so.6 (0x00007f3807e76000)
	libgmp.so.10 => /usr/lib/x86_64-linux-gnu/libgmp.so.10 (0x00007f3807bf5000)
	libxml2.so.2 => /usr/lib/x86_64-linux-gnu/libxml2.so.2 (0x00007f380783a000)
	libp11-kit.so.0 => /usr/lib/x86_64-linux-gnu/libp11-kit.so.0 (0x00007f38075d5000)
	libidn.so.11 => /usr/lib/x86_64-linux-gnu/libidn.so.11 (0x00007f38073a2000)
	libtasn1.so.6 => /usr/lib/x86_64-linux-gnu/libtasn1.so.6 (0x00007f380718f000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f3806f8a000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f3806d84000)
	libgraphite2.so.3 => /usr/lib/x86_64-linux-gnu/libgraphite2.so.3 (0x00007f3806b5d000)
	libdatrie.so.1 => /usr/lib/x86_64-linux-gnu/libdatrie.so.1 (0x00007f3806955000)
	libSM.so.6 => /usr/lib/x86_64-linux-gnu/libSM.so.6 (0x00007f380674d000)
	libICE.so.6 => /usr/lib/x86_64-linux-gnu/libICE.so.6 (0x00007f3806532000)
	libgomp.so.1 => /usr/lib/x86_64-linux-gnu/libgomp.so.1 (0x00007f3806310000)
	libgpg-error.so.0 => /lib/x86_64-linux-gnu/libgpg-error.so.0 (0x00007f38060fb000)
	libkrb5.so.3 => /usr/lib/x86_64-linux-gnu/libkrb5.so.3 (0x00007f3805e29000)
	libk5crypto.so.3 => /usr/lib/x86_64-linux-gnu/libk5crypto.so.3 (0x00007f3805bfa000)
	libcom_err.so.2 => /lib/x86_64-linux-gnu/libcom_err.so.2 (0x00007f38059f5000)
	libkrb5support.so.0 => /usr/lib/x86_64-linux-gnu/libkrb5support.so.0 (0x00007f38057ea000)
	libicuuc.so.55 => /usr/lib/x86_64-linux-gnu/libicuuc.so.55 (0x00007f3805456000)
	libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007f3805250000)
	libkeyutils.so.1 => /lib/x86_64-linux-gnu/libkeyutils.so.1 (0x00007f380504c000)
	libicudata.so.55 => /usr/lib/x86_64-linux-gnu/libicudata.so.55 (0x00007f3803594000)
```

**Comment by mathmanu** [comment]
I am using Ubuntu 14.04. In the CUDA Toolkit download page, I don't see an option for Ubuntu 14.04. Until CUDA8, Ubuntu 14.04 was supported. Do you know any work around?

**Comment by mathmanu** [comment]
One observation, there is no crash if I set MAX_PARALLEL_GROUPS to 1. But then its slightly slower (about 10% or so).
  static constexpr int MAX_PARALLEL_GROUPS = 1; //2;

Does this give any clue?

**Comment by drnikolaev** [comment]
@mathmanu thanks!
I'm using 14.04 with CUDA 9 perfectly fine.

Your new model worked ok until I started training in 16 bit mode. It crashed right away because of misaligned address. It means that 32-bit mode is most probably corrupted too. I'm working on the fix...

**Comment by mathmanu** [comment]
I have attached a picture of the latest CUDA download page. There is no link for Ubuntu14.04 currently.

<img width="822" alt="cuda9_download_page" src="https://user-images.githubusercontent.com/2990503/31488393-2207caa4-af5b-11e7-81da-c95a7bf21953.png">

Kindly let me know where to download CUDA9 for 14.04.

**Comment by mathmanu** [comment]
I got reply that CUDA 9 is not supported on Ubuntu 14.04
https://devtalk.nvidia.com/default/topic/1025050/cuda-setup-and-installation/is-cuda-9-available-for-ubuntu-14-04-for-x86_64-platform-/

I wonder how you managed to install it. Did you install the 16.04 deb file in 14.04 (it works some times) or is it a 14.04 installer that is not publicly released?

**Comment by drnikolaev** [comment]
hi @mathmanu I simply took the package and install it on my own risk. But seems like you can stay with 8.0. The corruption mentioned above has been fixed. Could you try it please?

**Comment by mathmanu** [comment]
Is it in drnikolaev/caffe?

**Comment by drnikolaev** [comment]
yep

**Comment by mathmanu** [comment]
@drnikolaev Tried it. It didn't help. The crash is still there.

**Comment by drnikolaev** [comment]
When it crashes, do you observe high temperature on your GPUs? Out of host memory? Anything else out of normal values?

**Comment by mathmanu** [comment]
@drnikolaev This occurs at an unpredictable iteration, so I have not been able to observe the system at the time of crash. However there is one thing - I am connecting to the machine remotely. When this crash occurs, I lose the connection.  When I log back in the program would have crashed. But the system behaves fine after that.

**Comment by mathmanu** [comment]
CUDA 9 was not installing in my 14.04 and I don't have enough knowledge and confidence to force it through. Hence I updated to the latest CUDA 8 and nvidia-375 with the following.

sudo apt-get purge nvidia*
sudo apt-get autoremove
sudo -E add-apt-repository ppa:graphics-drivers/ppa
sudo apt-get update
sudo apt-get install cuda
#now reboot the machine

I am running the training now, will let you know if it goes fine.

**Comment by mathmanu** [comment]
@drnikolaev is there some setting or some command to reduce the GPU utilization? Alternately how can I control the max GPU temperature at which the performance throttles? I want to try these if I still get a crash.

**Comment by drnikolaev** [comment]
Forgot to ask. Could you try to run caffe under `cuda-memcheck`?

**Comment by mathmanu** [comment]
No crash observed after the driver update. But I'll let it run more and see. 

Will run cuda-memcheck if I still get a crash. 

Thank you.

**Comment by mathmanu** [comment]
There was a program hang this time. I also tried running with cuda-memcheck, but even one iteration did not complete even after a long time.

With no other option left, I decided to again try the setting that I mentioned earlier:
**static constexpr int MAX_PARALLEL_GROUPS = 1; //2;**
There is no issue when I do this - there is no crash or hang. 

**And surprisingly in drnikolaev/caffe the speed is not reduced**. (There was about 10% speed reduction in NVIDIA/caffe for this network configuration when I set this flag to 1).

I am glad that there is a workaround.
